examples/app/stock_applet/stock_data.py

```diff
@@ -0,0 +1,49 @@
+import os
+import urllib
+import zipfile
+
+
+def download_data(url, save_dir, exclude_term):
+    name = os.path.join(save_dir, 'temp.zip')
+
+    #get the zip file
+    try:
+        name, hdrs = urllib.urlretrieve(url, name)
+    except IOError, e:
+      print "Can't retrieve %r to %r: %s" % (url, save_dir, e)
+      return
+
+    # open the zip file
+    try:
+        z = zipfile.ZipFile(name)
+    except zipfile.error, e:
+        print "Bad zipfile (from %r): %s"%(url, e)
+        return
+
+    # loop over files in list and extract each, if it isn't a directory
+    for n in z.namelist():
+
+        if not n.endswith('/'):
+            rel_file = n.replace(exclude_term, '')
+
+            dest = os.path.normpath(os.path.join(save_dir, rel_file))
+            destdir = os.path.dirname(dest)
+            if not os.path.isdir(destdir):
+                os.makedirs(destdir)
+            data = z.read(n)
+            f = open(dest, 'w')
```

Comment:
I would use a context manager here... but that's a personal choice... if you want to leave it as is, no problem...